components/server/src/server.rs

```diff
@@ -276,7 +276,7 @@ impl<ER: RaftEngine> TiKVServer<ER> {
 
         check_system_config(&config);
 
-        tikv_util::set_panic_hook(false, &config.storage.data_dir);
+        tikv_util::set_panic_hook(config.allow_core_dump, &config.storage.data_dir);
```

Comment:
maybe generate core dump only for the critical error which sets panic mark. For other panics, it may abort again and again, causing disk full eventually.

Reply:
Can we leave for the user to decide what's the best? For example, for rare memory corruption or segfault, no panic mark will be created, but there won't be crash loop as well.

Reply:
I believe the core dump is only used by us. When we want to investigate something, we let the user enable this config and wait for the core dump is generated.

Reply:
Since block cache is excluded from core dump, I think the core dump size would not be too large. There's still cases where we want to have core dump in cases other than panic mark. What do you think?

Reply:
okay